File: copl/com.elster.jupiter.domain.util/src/main/java/com/elster/jupiter/nls/Translation.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.nls;

import java.util.Locale;

public interface Translation {

    NlsKey getNlsKey();

    Locale getLocale();

    String getTranslation();
}

File: copl/com.elster.jupiter.loaderjs/src/main/web/js/ldr/src/model/Translation.js
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

/**
 * @class Ldr.model.Translation
 */
Ext.define('Ldr.model.Translation', {
    extend: 'Ext.data.Model',
    fields: [
        'cmp',
        'key',
        'value',
        {
            name: 'id',
            mapping: function (data) {
                return data.key + ':' + data.cmp
            }
        }
    ]
});